src/changes.rs

```diff
@@ -41,11 +41,12 @@ impl<'a> ChangeSet<'a> {
 
         for f in codemap.files.borrow().iter() {
             // Use the length of the file as a heuristic for how much space we
-            // need. I hope that at some stage someone rounds this up to the next
-            // power of two. TODO check that or do it here.
-            result.file_map.insert(f.name.clone(),
-                                   StringBuffer::with_capacity(f.src.as_ref().unwrap().len()));
+            // need. Round to the next power of two.
+            let file_size = f.src.as_ref().unwrap().len();
+            let power = (file_size as f64).log2().ceil() as u32;
+            let buffer_cap = 2usize.pow(power);
```

Comment:
You can do this far more efficiently with some bit twiddling: http://graphics.stanford.edu/~seander/bithacks.html#RoundUpPowerOf2